### PR TITLE
[Impeller] remove most temporary allocation during polyline generation.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Part of https://github.com/flutter/flutter/issues/143077

Only allocate into reused arenas instead of allocating a new vector of data. Fixes https://github.com/flutter/flutter/issues/133348

Also moves tessellation logic into the c/tessellator and out of the impeller tessellator. This was necessary to fix a compilation error. introduced by including host_buffer -> allocator -> fml mapping -> window.h include which has a function definition that conflicts with the c tessellator definition.

### CODE REVIEW

File: impeller/tessellator/tessellator_unittests.cc [hunk 1]
```diff
@@ -83,26 +83,34 @@ TEST(TessellatorTest, TessellatorBuilderReturnsCorrectResultStatus) {
...
-    auto pts = t.TessellateConvex(
-        PathBuilder{}.AddRect(Rect::MakeLTRB(0, 0, 10, 10)).TakePath(), 1.0);
-
-    std::vector<Point> expected = {{0, 0}, {10, 0}, {0, 10}, {10, 10}};
-    EXPECT_EQ(pts, expected);
+    t.TessellateConvexInternal(
+        PathBuilder{}.AddRect(Rect::MakeLTRB(0, 0, 10, 10)).TakePath(), points,
+        indices, 1.0);
+
+    std::vector<Point> expected = {{10, 0}, {10, 10}, {0, 10}, {0, 0}};
```
Comment (jonahwilliams) [inline impeller/tessellator/tessellator_unittests.cc:93]: There order of these changes because we hit the "first" point in the contour on close now.

**Comment by jonahwilliams** [comment]
The only golden is a minor trivial diff:

image

**Review by chinmaygarde — APPROVED** [review]

**Comment by jonahwilliams** [comment]
reason for revert: breaking flutter logo rendering

image

**Comment by jonahwilliams** [comment]
This was probably relying on us closing paths? or pehaps this is messing up unclosed paths that are implicitly closed. At any rate this is a good example to go off of.

**Pull request merged**